public/components/trace_analytics/requests/queries/traces_queries.ts

```diff
@@ -205,10 +363,28 @@ export const getSpanDetailQuery = (traceId: string, size = 3000) => {
       ],
     },
   };
-  return query;
 };
 
-export const getPayloadQuery = (traceId: string, size = 1000) => {
+export const getPayloadQuery = (mode: TraceAnalyticsMode, traceId: string, size = 1000) => {
```

Comment:
Seems like the query returned from getPayloadQuery shares the very similar with getSpanFlyoutQuery. Then can we extract them and create a base function maybe like
```
getBaseQuery = ({ traceKey, traceId, ... }) => {
  return {
    ...
    term: {
                [traceKey]: traceId,
              },
  }
};
```

Reply:
Not fully understanding - but I was looking for a way to dynamically change the key type - is this a way how? However even if it is, still wondering what the best path here is - should we just keep separate queries to make things very clear to we can reuse queries and dynamically create key/value mapping?

Reply:
I feel it depends on how often we change the query structure for different cases and how many / what are the parts we are changing. It seems to me that at this moment, the only difference is the term part, then we should have a query template and only changes the term. But I think it also makes sense here with the notion of separate queries to make things clear, so one option here i'm thinking is maybe we can remove `if (mode === 'jaeger')` and create separate functions for jaeger and non jaeger cases to be something like

```
getBaseQuery - ({ term, any other... }) => {}
getJaegerPayloadQuery = () => {
  // do jaeger related processing
  // get query template or have its own query 
return getBaseQuery(jaegerParams);
}
getDataPrepperPayloadQuery = () => {
  // do data prepper related processing
 // get query template or have its own query 
return getBaseQuery(prepperParams);
}
```
I feel with this change, we could share query if they share the same or use different query structures if they diverged a lot. I may be wrong as I may miss some use cases for trace analytics or you may have better options or thoughts.